endaq/batch/analyzer.py

```diff
@@ -24,47 +28,80 @@
+    """
+
+    accel_highpass_cutoff: Optional[float]
+    accel_integral_tukey_percent: float
+    accel_integral_zero: typing.Literal["start", "mean", "median"]
+    accel_start_time: Optional[np.timedelta64]
+    accel_end_time: Optional[np.timedelta64]
+    accel_start_margin: Optional[np.timedelta64]
+    accel_end_margin: Optional[np.timedelta64]
+    psd_freq_bin_width: float
+    psd_window: str
+    pvss_init_freq: float
+    pvss_bins_per_octave: float
+    vc_init_freq: float
+    vc_bins_per_octave: float
+
+
+class CalcCache:
     """
     A wrapper for `idelib.dataset.Dataset` that caches channel data streams as
     `pandas.Dataset`s.
     """
 
     PV_NATURAL_FREQS = np.logspace(0, 12, base=2, num=12 * 12 + 1, endpoint=True)
 
-    def __init__(
-        self,
-        dataset,
-        *,
-        preferred_chs=[],
-        accel_highpass_cutoff,
-        accel_integral_tukey_percent,
-        accel_integral_zero,
-        accel_start_time,
-        accel_end_time,
-        accel_start_margin,
-        accel_end_margin,
-        psd_freq_bin_width,
-        psd_window="hanning",
-        pvss_init_freq,
-        pvss_bins_per_octave,
-        vc_init_freq,
-        vc_bins_per_octave,
-    ):
+    def __init__(self, data, params: CalcParams):
         """
         Copies out the numpy arrays for the highest priority channel for each
         sensor type, and any relevant metadata.  Cuts them into chunks.
         """
-        if accel_start_time is not None and accel_start_margin is not None:
+        if (
+            params.accel_start_time is not None
+            and params.accel_start_margin is not None
+        ):
             raise ValueError(
                 "only one of `accel_start_time` and `accel_start_margin` may be set at once"
             )
-        if accel_end_time is not None and accel_end_margin is not None:
+        if params.accel_end_time is not None and params.accel_end_margin is not None:
             raise ValueError(
                 "only one of `accel_end_time` and `accel_end_margin` may be set at once"
             )
 
-        self._channels = ide_utils.dict_chs_best(
+        self._channels = data
+
+        self._accel_highpass_cutoff = params.accel_highpass_cutoff
+        self._accel_start_time = params.accel_start_time
+        self._accel_end_time = params.accel_end_time
+        self._accel_start_margin = params.accel_start_margin
+        self._accel_end_margin = params.accel_end_margin
+        self._accel_integral_tukey_percent = params.accel_integral_tukey_percent
+        self._accel_integral_zero = params.accel_integral_zero
+        self._psd_window = params.psd_window
+        self._psd_freq_bin_width = params.psd_freq_bin_width
+        self._pvss_init_freq = params.pvss_init_freq
+        self._pvss_bins_per_octave = params.pvss_bins_per_octave
+        self._vc_init_freq = params.vc_init_freq
+        self._vc_bins_per_octave = params.vc_bins_per_octave
```

Comment:
Is there a reason to store these individually?  I feel like it would be safer to save it as `self._params = params.copy()` so that if `params` gets new fields in the future you don't have to worry about adding them to this block of code.

Reply:
nope, just an artifact of the old way the code was written. good call, thanks! changed in 0b923ca40abe255e704bcc17ffd9c578102bc870